include/drivers/spi.h

```diff
@@ -129,11 +130,17 @@ extern "C" {
  *    to act as a CS line
  * @param delay is a delay in microseconds to wait before starting the
  *    transmission and before releasing the CS line
+ * @param gpio_dt_flags is the devicetree flags corresponding to how the CS
+ *    line should be driven. GPIO_ACTIVE_LOW and GPIO_ACTIVE_HIGH are ignored
+ *    in favor of SPI_CS_ACTIVE_HIGH and SPI_CS_ACTIVE_LOW options in struct
```

Comment:
May need to update if devicetree flags are respected as suggested above.

Reply:
The original documentation was to that effect. From @tbursztyka's review I gathered transitioning to the devicetree flags for active level wouldn't be possible due to non GPIO CS controllers

Reply:
See my comment below about an assert that would check SPI_CS_ACTIV_* vs GPIO_ACTIVE_* compliance